Reference_Manual.md

```diff
@@ -3287,6 +3294,245 @@ This typed_data allows you to retrieve the data type of the underlying binary to
+### 7.11.1 Overview
+All the sql related types and functionalilities accomodates in `hazelcast::client::sql` namespace. It is possible to execute queries on `IMap`, `Kafka` and `Files` by using this module. 
+
+`sql_service` is the main controller of this module. Queries can be executed via `sql_service` class. It also supports to specify query parameters. After executing any query `boost::future<std::shared_ptr<sql_result>>` is returned.
+
+If it is a `SELECT` query then it provides a way to iterate over rows. Rows are stored in pages and essentially, hazelcast serves result of `SELECT` queries page by page so the interface reflects this approach.
+
+`sql_result` provides an `page_iterator` which allows caller to fetch and iterate over pages. `sql_result::page_iterator::next()` returns `boost::future<std::shared_ptr<sql_page>>`. Every `sql_page` contains either `0` or more rows. It also tells that whether this page is the `last` or not via `last()` method.
+### 7.11.2 Create Mapping
+Before you can access any object using SQL, a mapping has to be created. See the documentation for the [CREATE MAPPING](https://docs.hazelcast.com/hazelcast/5.1/sql/create-mapping) command.
+
+``` C++
+using namespace hazelcast::client::sql;
+
+auto hz = hazelcast::new_client().get();
+
+// populate the map with some data
+auto map = hz.get_map("integers").get();
+for (int i = 0; i < 1000; ++i) {
+    map->put(i, i).get();
+}
+
+auto sql = hz.get_sql();
+// Create mapping for the integers. This needs to be done only once per map.
+auto result = sql.execute(R"(
+                CREATE OR REPLACE MAPPING integers
+                    TYPE IMap
+                    OPTIONS (
+                        'keyFormat' = 'int',
+                        'valueFormat' = 'int'
+                        )
+                    )").get();
+```
+### 7.11.3 Read SELECT Results
+`SELECT` results are fetched page by page. A page can accomodate zero or more rows.
+
+`sql_service::execute` executes the query and returns a `boost::future<std::shared_ptr<sql_result>>`. Via `sql_result::iterator()` method, `sql_result::page_iterator` is acquired and pages can be iterated.
+``` C++
+using namespace hazelcast::client::sql;
+
+auto hz = hazelcast::new_client().get();
+
+auto result = hz.get_sql().execute("SELECT * FROM integers").get();
+
+result->row_metadata(); // Tells about metadata
+
+for (auto itr = result->iterator(); itr.has_next();)
+{
+    auto page = itr.next().get();
+
+    for(const sql_page::sql_row& row : page->rows())
+    {
+        std::cout << row.get_object<int>(0)
+                  << ", "
+                  << row.get_object<int>(1)
+                  << '\n';
+    }
+}
+```
+
+**Note-1**: First call to `sql_result::page_iterator::next()` function will return `ready` future because first page is already loaded.
```

Comment:
```suggestion
**Note-1**: First call to `sql_result::page_iterator::next()` function will return `ready` future because first page is already loaded as a result of the sql_service::execute call.
```

Reply:
Fixed.